src/activity_handlers.cpp

```diff
@@ -1501,6 +1502,51 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
     act->moves_left = actor->move_cost;
 }
 
+void activity_handlers::mend_item_finish( player_activity *act, player *p )
+{
+    item_location target;
+    switch( static_cast<item_location::type>( act->index ) ) {
```

Comment:
The rest of the PR checks out, but I'm not ok merging this part of the change, it's really hard to put the genie back in the bottle once it's out, especially since in this case it involves writing the player_activity to a savefile.  We need to get that part out of the way up front so we don't have to patch in conversion code.

I'm working on this feature right now, it should be ready today or tomorrow.